app/quadraticlands/views.py

```diff
@@ -295,3 +298,129 @@ def error(request, code):
+                    )
+                game = create_game_helper(request.user.profile.handle, title)
+                messages.success(
+                    request,
+                    f'Your Game has been created successfully'
+                )
+                return redirect(game.url)
+    else:
+        if request.POST:
+            messages.error(
+                request,
+                f'You must login to create a game.'
+            )
+
+
+
+    # return the vite
+    params = {
+        'title': 'Quadratic Diplomacy',
+        'games': games,
+        'invite': invited_to_game,
+    }
+    return TemplateResponse(request, 'quadraticlands/mission/diplomacy/index.html', params)
+
+
+@login_required
+def mission_diplomacy_room(request, uuid, name):
+    # get the game
+    try:
+        game = Game.objects.get(uuid=uuid)
+    except:
+        raise Http404
+    return mission_diplomacy_room_helper(request, game)
+
+def mission_diplomacy_room_helper(request, game):
+
+    # handle invites
+    players = GamePlayer.objects.filter(active=True, profile=request.user.profile)
+    games = Game.objects.filter(pk__in=players.values_list("game"))
+    if game not in games:
+        if request.user.is_authenticated:
+
+            # too many user condition
+            if game.active_players.count() >= max_players_per_game:
+                messages.info(
+                    request,
+                    f'There are already {max_players_per_game} in the game.  Wait for someone to leave + try again.'
+                )
+                return mission_diplomacy_helper(request)
+
+            game.add_player(request.user.handle)
+            return mission_diplomacy_helper(request, invited_to_game=game)
+        else:
+            # logged out condition
+            messages.info(
+                request,
+                f'Please login to join this game.'
+            )
+
+    # in game experience
+    is_member = game.is_active_player(request.user.profile.handle)
+    is_admin = players.filter(admin=True, profile=request.user.profile).exists()
+    if is_admin:
+        if request.GET.get('remove'):
+            remove_this_fool = request.GET.get('remove')
+            for player in players.filter(active=True, profile=remove_this_fool):
+                game.remove_player(player.profile.handle)
+                messages.info(
+                    request,
+                    f'{remove_this_fool} has been removed'
+                )
+    
+    # delete game
+    if is_member and request.GET.get('delete'):
+        game.remove_player(request.user.profile.handle)
+        messages.info(
+            request,
+            f'You have left this game.'
+        )
+        return redirect('/quadraticlands/mission/diplomacy')
+
+    # chat in game
+    if is_member and request.POST.get('chat'):
+        game.chat(request.user.profile.handle, request.POST.get('chat'))
+
+    # make a move
+    if is_member and request.POST.get('signature'):
```

Comment:
yooo @thelostone-mc any idea how i can get this POST dict as a nested dict

eg right now its 

```
{
   "action":"vouched",
   "stakes":{
      "signature":"0x2075189c5fab8023869bc64135f38d8dc44666ec435fc6b8a95d38063afb9775311550fa70af7275a0dab3b8442ccec79026cc378f730cb83bf3cc42501f0b361b",
      "package[account]":"0x00De4B13153673BCAE2616b67bf822500d325Fc3",
      "package[balance]":"0",
      "csrfmiddlewaretoken":"uny5ZizizmYtcZJCB9WNycevCaPkkzYr0kLVFxnACqOqdoPo6txrzQEItFUWgYDf",
      "package[votes][0][value]":"",
      "package[votes][1][value]":"",
      "package[votes][2][value]":"333",
      "package[votes][0][userid]":"1",
      "package[votes][1][userid]":"3",
      "package[votes][2][userid]":"2",
      "package[votes][0][username]":"owocki",
      "package[votes][1][username]":"octavioamu",
      "package[votes][2][username]":"thelostone-mc"
   }
}
```

but i want it to look like
```
{
   "action":"vouched",
   "stakes":{
      "signature":"0x2075189c5fab8023869bc64135f38d8dc44666ec435fc6b8a95d38063afb9775311550fa70af7275a0dab3b8442ccec79026cc378f730cb83bf3cc42501f0b361b",
      "package[account]":"0x00De4B13153673BCAE2616b67bf822500d325Fc3",
      "package[balance]":"0",
      "csrfmiddlewaretoken":"uny5ZizizmYtcZJCB9WNycevCaPkkzYr0kLVFxnACqOqdoPo6txrzQEItFUWgYDf",
      "package": {
                  "key1": ......,
                  "key2": ......,
                  "key3": ......,
      },
   }
}
```

Reply:
Hmm would key1 be username , key2 be userid and key3 be value ?

Reply:
fixed